### PR TITLE
NIFI-13943: Upgrade Dropbox, Salesforce, SNMP, and QuestDB depdendencies

### PR DESCRIPTION
**Pull request by mattyb149**
# Summary

[NIFI-13943](https://issues.apache.org/jira/browse/NIFI-13943) This PR upgrades some libraries that support EOL products

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-extension-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/pom.xml [hunk 1]
```diff
@@ -42,7 +42,7 @@
         <dependency>
             <groupId>org.apache.camel</groupId>
             <artifactId>camel-salesforce</artifactId>
-            <version>3.14.10</version>
+            <version>3.22.2</version>
```
Comment (joewitt) [inline nifi-extension-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/pom.xml:45]: Notably 3.22.x is EOL in Dec 2024 (one month).

File: nifi-extension-bundles/nifi-dropbox-bundle/pom.xml [hunk 2]
```diff
@@ -27,7 +27,7 @@
     <packaging>pom</packaging>
 
     <properties>
-        <dropbox.client.version>5.4.6</dropbox.client.version>
+        <dropbox.client.version>7.0.0</dropbox.client.version>
```
Comment (joewitt) [inline nifi-extension-bundles/nifi-dropbox-bundle/pom.xml:30]: reading their changelog in github - this one 'seems' safe

File: nifi-extension-bundles/nifi-snmp-bundle/nifi-snmp-processors/pom.xml [hunk 3]
```diff
@@ -20,8 +20,8 @@ language governing permissions and limitations under the License. -->
     <packaging>jar</packaging>
 
     <properties>
-        <snmp4j.version>2.8.18</snmp4j.version>
-        <snmp4j-agent.version>2.7.9</snmp4j-agent.version>
+        <snmp4j.version>3.8.2</snmp4j.version>
+        <snmp4j-agent.version>3.8.1</snmp4j-agent.version>
```
Comment (joewitt) [inline nifi-extension-bundles/nifi-snmp-bundle/nifi-snmp-processors/pom.xml:24]: reviewing if changes seem safe https://www.snmp4j.org/CHANGES.txt but seems like the 2.x line was Java 8 and the 3.x line is Java 9 or higher

File: nifi-framework-bundle/nifi-framework-extensions/nifi-questdb-bundle/nifi-questdb/pom.xml [hunk 4]
```diff
@@ -35,7 +35,7 @@
         <dependency>
             <groupId>org.questdb</groupId>
             <artifactId>questdb</artifactId>
-            <version>7.4.2</version>
+            <version>8.1.4</version>
```
Comment (joewitt) [inline nifi-framework-bundle/nifi-framework-extensions/nifi-questdb-bundle/nifi-questdb/pom.xml:38]: reviewing release notes https://github.com/questdb/questdb/releases

**Review by exceptionfactory — APPROVED** [review]
Thanks for the upgrades and adjustments @mattyb149, and thanks for reviewing the release notes @joewitt. This set of upgrades looks good. +1 merging

**Pull request merged**